lib/compat/wordpress-6.4/fonts/font-face/class-wp-font-face-resolver.php

```diff
@@ -71,12 +71,7 @@ private static function parse_settings( array $settings ) {
 						continue;
 					}
 
-					// Prepare the fonts array structure for this font-family.
-					if ( ! array_key_exists( $font_family_name, $fonts ) ) {
-						$fonts[ $font_family_name ] = array();
-					}
-
-					$fonts[ $font_family_name ] = static::convert_font_face_properties( $definition['fontFace'], $font_family_name );
+					$fonts[] = static::convert_font_face_properties( $definition['fontFace'], $font_family_name );
```

Comment:
This should move to 6.5 folder now I think.

Reply:
Do you think that because of this particular change or because of something else?

If you mean because of this particular change, this is my rationale:

I understand that the `wordpress-6.4` folder holds the code needed to make the code work with the latest Gutenberg version running on WordPress < 6.4, right? That's still true, so I'm unsure why this should be moved.

Reply:
> wordpress-6.4 is the code that was backported during 6.4, so to make Gutenberg work with 6.3 and lower.

I think we said the same but with different words.

 ---
 
Given this PR is only a bugfix, and these classes were merged in `6.4,` and only one bugfix will be different in `6.5`, should we still move all files to the WordPress-6.5 folder?

Reply:
yes, because the 6.4 folder will get removed once the minimum supported version for Gutenberg is going to be 6.4 but that particular fix will still be needed.

Reply:
In other words, after a WP version is released, we shouldn't touch its dedicated folder on Gutenberg anymore (if not for moving them to a more recent folder to be adapted)

Reply:
Thanks for the explanation, I moved the files in this commit: https://github.com/WordPress/gutenberg/pull/59376/commits/2ec7d76da24297ce398a97007d391adc7ef4a3da

Reply:
What about the current file here, should we remove it.

Also, something that is not clear is that in 6.4, this class is defined in WordPress, which means the Gutenberg class won't be used which means the bug will remain when you combine Gutenberg + 6.4.

Reply:
> means the bug will remain when you combine Gutenberg + 6.4.

Good point. Should we merge this PR first and work with these changes on the Gutenberg-specific class (`Gutenberg_Font_Face_Resolver_6_4`) that PR is adding? Otherwise, I don't see a way of overriding in Gutenberg plugin the WP_Font_Face_Resolver class existing in WordPress 6.4.

Reply:
I think if we don't figure out a way to override the 6.4 use-case properly, then there's no match point in "fixing" this bug in Gutenberg. Maybe we should just close this PR.

Reply:
> figure out a way to override the 6.4 use-case properly

The use 6.4 use-case would be printing all the font faces when running wp 6.4 + the latest GB plugin, but I don't understand the quote. Could you explain or add more context, please?

Reply:
I don't understand your question :P 

I'm saying that if we can't override the class properly on Gutenberg, why change it at all, the 6.4 code should match the code that was merged in 6.4 and we can't move it to 6.5 and update it because it will have no effect anyway.

Reply:
Oh, I see, thanks for explaining. In that case, the bug will be still there when running 6.4 + Gutenberg. Is that ok? If that's OK we can close the PR.

Reply:
It's ok for me I think.